### PR TITLE
Pie 2475 - Key Vault Caching

### PR DESCRIPTION
**Pull request by hgreene933**
This PR introduces caching for Key Vault secrets. This should help us avoid being rate limited and incurring large Azure fees. The caching is powered via APCU and follows the PSR-6 Caching Interface standard. Composer packages were also updated in order to fix dependabot complaints.

### CODE REVIEW

File: src/Secret.php [hunk 1]
```diff
@@ -54,12 +57,27 @@ public function getSecretVersions(string $secretName) {
...
+    {
+        // Set cache location
+        $cache = new ApcuCachePool();
+
         if ($secret instanceof SecretVersionEntity && !$secretVersion) {
             $secretVersion = $secret->id;
             $secret = $secret->name;
         }
 
+        $wordpressSecretVersion = $_ENV['WORDPRESS_SECRET_VERSION'] ?? '';
```
Comment (nickmblain) [inline src/Secret.php:70]: Should we add something about this config value to the readme?

Comment (hgreene933) [reply src/Secret.php:70]: That's, like, so much work. but fine I've added that bad boy.

File: src/Secret.php [hunk 2]
```diff
@@ -83,14 +102,22 @@ public function getSecret($secret, string $secretVersion = null) {
             ),
             $response->contentType ?? null,
         );
+
+        // Set the secret to expire in one day and store it in cache
+        $secretCache->expiresAt(new \DateTime('next month'));
+        $secretCache->set($secretEntity);
+        $cache->save($secretCache);
```
Comment (mcurtisCMNH) [inline src/Secret.php:109]: do you know if this is encrypted at rest?

File: src/Secret.php [hunk 3]
```diff
@@ -125,26 +152,27 @@ public function getSecrets(string $nextLink = null): IdRepository {
...
+     * If the named secret already exists, Azure Key Vault creates a new version of that secret.
      * @param string $secretName
      * @param string $value
      * @param SecretAttributeEntity|null $secretAttributes
      * @param string|null $contentType
      * @param array|null $tags
      * @return SecretEntity
      */
-    public function setSecret(string $secretName, string $value, $secretAttributes = null, string $contentType = null, array $tags = null) {
+    public function setSecret(string $secretName, string $value, $secretAttributes = null, string $contentType = null, array $tags = null)
```
Comment (mcurtisCMNH) [inline src/Secret.php:163]: What secrets do we need to set programmatically like this?

**Review by nickmblain — APPROVED** [review]

File: src/Secret.php (hunk 3 above)
Comment (hgreene933) [reply src/Secret.php:163]: None, but this is forked from a library that included all of this logic. So there's several methods that we'll probably never use, but it's not worth going through and removing them all.

File: src/Secret.php (hunk 2 above)
Comment (hgreene933) [reply src/Secret.php:109]: It is not, but neither was the original implementation.

File: src/Secret.php (hunk 3 above)
Comment (mcurtisCMNH) [reply src/Secret.php:163]: What do you think about removing the set specifically then? That way we don't have to give the site write permissions in keyvault and reduce our attack surface.

File: src/Secret.php (hunk 2 above)
Comment (mcurtisCMNH) [reply src/Secret.php:109]: It kind of invalidates the whole point of keyvault to not encrypt it... perhaps we could explore another caching solution that isn't on the filesystem?

Comment (mcurtisCMNH) [reply src/Secret.php:109]: I misunderstood - this isn't a file cache it's an in-memory cache. I'm cool with that.

File: src/Secret.php (hunk 3 above)
Comment (mcurtisCMNH) [reply src/Secret.php:163]: On second thought we could just not give the applications write access and it doesn't matter if we have this code or not.

**Review by mcurtisCMNH — APPROVED** [review]

**Pull request merged**